### DISCUSSION TITLE
Dictionary opening/closing brace on a separate lines?

### DISCUSSION THREAD

**Discussion by darktempla** [astral-sh/ruff#13013] [General]
Any guidance on this would be appreciated. 

When running `ruff format --diff` why is the dictionary opening/closing braces being put on a new line?

Apart from a `# fmt: off` block is there some way via the configuration to disable this formatting? 

```
-    context.add_output_metadata({
-        "file_name": report.file_name,
-        "sheet_names": list(report.workbook_sheets.keys()),
-        "sheet_count": len(report.workbook_sheets.keys()),
-    })
+    context.add_output_metadata(
+        {
+            "file_name": report.file_name,
+            "sheet_names": list(report.workbook_sheets.keys()),
+            "sheet_count": len(report.workbook_sheets.keys()),
+        }
+    )
```

It doesn't format the dictionary if it is assigned to a variable only when an inline function parameter.

```
# not formatted

meta_data = {
    "file_name": report.file_name,
    "sheet_names": list(report.workbook_sheets.keys()),
    "sheet_count": len(report.workbook_sheets.keys()),
}
```

**Configuration (pyproject.toml)**
```
[tool.ruff]
line-length = 160
target-version = "py311"
src = ["my_module", "tests"]


[tool.ruff.lint]
select = ["ALL"]
ignore = [
    "ANN001", "ANN101", "ANN201",
    "COM812",
    "D100", "D101", "D102", "D103", "D104", "D107", "D200", "D213", "D202", "D203", "D407",
    "FIX002", "G004", "ISC001",
    "PD002", "PD901", "PLR0913", "PT009",
    "SIM201", "S301",
    "TD002", "TD003", "T201"
]
```

**Comment by MichaReiser** [comment]
Hi @darktempla 

The formatting you're observing is the current stable style. Ruff supports a new formatting style in preview-mode that ensures that dictionary and other parenthesized expressions in single-argument call expressions keep the parentheses on the same line. See https://github.com/astral-sh/ruff/issues/8279

You can opt-in to preview styles by setting `tool.ruff.format.preview = true`. But be warned that you not only opt into using that specific preview style, but all preview styles and preview styles can change even in patch releases.

**Reply by darktempla** [reply]
Thanks @MichaReiser.

I was trying to figure out if there was some way with the formatter to trace it back to a rule or otherwise that was causing this to reduce my search space. Perhaps it was late or perhaps my google search skills are lacking but I didn't stumble across the issue you highlighted. So thanks for pointing it out. Makes sense.

I have confirmed adding the following to `pyproject.toml`
```
[tool.ruff.format]
preview = true
```
Keeps the `({` braces paired together.

**Reply by MichaReiser** [reply]
I think finding that issue with google is difficult. It uses very specific language. I wouldn't find it. I only found it because I know the labels and preview feature name.
